### ISSUE TITLE
Invalid QR Code with URL query string

### ISSUE THREAD

**Issue by fencer** [codebude/QRCoder#340]
## Type of issue
[x ] Bug  
[ ] Question (e.g. about handling/usage)  
[ ] Request for new feature/improvement  

I am generating a URL QR code using the code below. When the URL is set to the value in `workingURL`, everything works fine. When URL  is set to `nonWorkingURL` (same value but with a query string) the resulting QR image is not readable.

```
string workingURL = "https://github.com/";
string nonWorkingURL = "https://github.com?a=b&x=y";
Url generator = new Url(nonWorkingURL);
string payload = generator.ToString();
QRCodeData qrCodeData = qrGenerator.CreateQrCode(payload, QRCodeGenerator.ECCLevel.M, requestedVersion: 4);
QRCode qrCode = new QRCode(qrCodeData);
Image qrImage = Image.GetInstance(qrCode.GetGraphic(2), BaseColor.White, true);
```
**Invalid QR:**
![image](https://user-images.githubusercontent.com/620966/140832853-41d93437-0e2c-47d3-9b7d-b33650fc9c70.png)

Valid QR:
![image](https://user-images.githubusercontent.com/620966/140832950-2467c658-b468-4167-8b23-229c3a1277d8.png)

## Your Environment
IIS Server on Windows 10 
.net5.0
QRCoder 1.4.1 Installed from Nuget.

**Comment by codebude** [comment]
Hi @fencer ,

can you send the exact url that you want to encode and that broke?
From your sample QR codes I got `https://localhost:44369/apple-icam?utm_source=qrcode&utm_mediu`
I guess it should be `https://localhost:44369/apple-icam?utm_source=qrcode&utm_medium`, correct?

**Comment by codebude** [comment]
Ok, got it. The url gets cut, because it is longer/takes more bytes than the requested QR code version (=4) can save. Thus the urls "break".
This topic is already fixed by this commit: https://github.com/codebude/QRCoder/commit/49d5b35e186437c086de152f50a0c7348f7c84cc
The changes will be part of the next QRCoder release (1.4.2). In 1.4.2 an exception like `The given payload exceeds the maximum size of the QR code standard. The maximum size allowed for the choosen paramters (ECC level=M, EncodingMode=Byte, FixedVersion=4) is 62 byte.` will be thrown.

**TL;DR;** The error handling in 1.4.1 doesn't warn you, if requested version is smaller than the data you like to encode.
Possible solutions:
- [Option A] Remove "requestedVersion" parameter and let QRCoder choose the optimal version. (QRCoder will choose the smallest version that can handle your payload string.)
- [Option B] Manually Set a "requestedVersion" that is big enough to handle/save your payload completely. (This table shows you how much bytes (column: data bits) you can store in which version depending on the choosen error correction level.)
